src/basic/process-util.c

```diff
@@ -1199,12 +1199,12 @@ int pid_from_same_root_fs(pid_t pid) {
 }
 
 bool is_main_thread(void) {
-        static thread_local int cached = 0;
+        static thread_local int cached = -1;
 
-        if (_unlikely_(cached == 0))
-                cached = getpid_cached() == gettid() ? 1 : -1;
+        if (_unlikely_(cached < 0))
```

Comment:
might as well drop the `_unlikely_()` here, afaics there's a good chance we call this function once only in each tool, which hence means this optimization won't help but likely make things worse...

Reply:
-> #35561